features/shopping_lists/src/main/kotlin/gq/kirmanak/mealient/shopping_lists/ui/details/ShoppingListScreen.kt

```diff
@@ -388,18 +419,26 @@ private fun ShoppingListItemEditorFoodRow(
                 colors = ExposedDropdownMenuDefaults.outlinedTextFieldColors(),
             )
 
-            ExposedDropdownMenu(
+            val foodFilteringOptions = state.foods.filter {
+                it.name.contains(foodSearchQuery, ignoreCase = true)
+            }
```

Comment:
```suggestion
            val foodFilteringOptions = remember(state.foods, foodSearchQuery) {
                state.foods.filter {
                    it.name.contains(foodSearchQuery, ignoreCase = true)
                }
            }
```

The result of this calculation should be remembered, otherwise the filter will be executed very often causing poor performance. Please, do the same for the units.